app/javascript/components/request-workflow-status/index.jsx

```diff
@@ -0,0 +1,125 @@
+import React, { useEffect, useState } from 'react';
+import {
+  Tag, Loading,
+} from 'carbon-components-react';
+import PropTypes from 'prop-types';
+import { workflowStatusData, workflowStateTypes } from './data';
+import MiqDataTable from '../miq-data-table';
+import NotificationMessage from '../notification-message';
+
+/** Component to render the Workflow status in /miq_request/show/#{id} page */
+const RequestWorkflowStatus = ({ recordId }) => {
+  const RELOAD = 2000; // Time interval to reload the RequestWorkflowStatus component.
+
+  const [data, setData] = useState(
+    {
+      isLoading: true,
+      responseData: undefined,
+      message: undefined,
+    }
+  );
+
+  /** Function to get the Workflow
+   * TODO: Use the recordId in the api url when it's ready.
+   */
+  const getWorkflow = () => {
+    const url = `/api/configuration_script_payloads/${recordId}`; // TODO: the usage of recordId is to be changed...
```

Comment:
@jeffibm are you trying to show the workflow_instance (aka the running one) here?  Because `/api/configuration_script_payloads` will give you the workflow definition.

Reply:
which API should we use?

Reply:
`api/configuration_scripts` ?

Reply:
Yes

Reply:
`context["States"]` was renamed to `context["StateHistory"]` a long time ago.  If you don't have anything in StateHistory that means your workflow_instance isn't actually running.

Reply:
I used `State` to render the list before.
so I can replace `State` with `StateHistory` ?

Reply:
> I used State to render the list before.

State?  Or States?

State is the current state, StateHistory (previously States) is the list

Reply:
sorry for the typo, I used '`States`' before.
Shall I replace `States` with `StateHistory` now?